src/Component/Product/ProductListContainer.tsx

```diff
@@ -1,58 +1,29 @@
-import { useState, useEffect } from 'react';
-import { ProductTypes } from '../../types/ProductTypes';
 import ProductList from './ProductList';
 import ProductListToolbar from './ProductListToolbar';
 import getProducts from '../../api/getProducts';
-import { CartItemTypes } from '../../types/CartItemType';
 import Spinner from '../Common/Spinner';
 import styled from '@emotion/styled';
+import { useAPIContext } from '../Common/Provider';
 
 interface ProductListContainerProps {
-  cartItems: CartItemTypes[];
-  updateCartItems: () => void;
-  getMatchCartItem: (id: number) => CartItemTypes | undefined;
   updateErrorMessage: (errorMessage: string) => void;
-  checkMax: () => boolean;
 }
 
-type Status = 'idle' | 'loading' | 'success' | 'error';
-
 export default function ProductListContainer({
-  updateCartItems,
-  getMatchCartItem,
   updateErrorMessage,
-  checkMax,
 }: ProductListContainerProps) {
-  const [products, setProducts] = useState<ProductTypes[]>([]);
-  const [status, setStatus] = useState<Status>('idle');
-
-  useEffect(() => {
-    async function fetchProducts() {
-      try {
-        setStatus('loading');
-        const productsData = await getProducts();
-        const productsContent = productsData.content;
-        setProducts(productsContent);
-        setStatus('success');
-      } catch (e) {
-        setStatus('error');
-        updateErrorMessage(
-          ' 상품 목록 조회 중 오류가 발생했습니다. 잠시 후 다시 시도해 주세요.'
-        );
-      }
-    }
-    fetchProducts();
-  }, [updateErrorMessage]);
+  const { data: products, status } = useAPIContext({
+    apiFn: () => getProducts(),
+    key: 'products',
+  });
```

Comment:
다음과 같은 훅을 통해 관리 포인트를 일원화 할 수 있을 것으로 보입니다.

```suggestion
  const useProducts = () => useAPIContext({
    apiFn: () => getProducts(),
    key: 'products',
  });
  
  ...
  
  import useProducts from '~~'

  const { data: products, status } = useProducts();
```

Reply:
이렇게 하면 key값이 헷갈리는 일도 없고 api를 적절하지 못한 것을 넘겨주는 실수를 방지할 수 있겠네요!!

Reply:
수정 커밋: [refactor: useProducts 훅을 이용해 useAPIContext 사용하던 부분 변경](https://github.com/woowacourse/react-shopping-products/pull/153/commits/544e0aead104b55abaa3cd24a599c2b34c4bf55b)